GenderPayGap.WebUI/Controllers/Admin/AdminOrganisationReturnController.cs

```diff
@@ -0,0 +1,96 @@
+
+            return View("ViewReturns", organisation);
+        }
+
+        [HttpGet("organisation/{id}/returns/details")]
+        public IActionResult ViewReturnDetails(long id)
+        {
+            Organisation organisation = dataRepository.Get<Organisation>(id);
+
+            return View("ViewReturnDetails", organisation);
+        }
+
+        [HttpGet("organisation/{id}/returns/download-details-csv")]
+        public IActionResult DownloadReturnDetailsCsv(long id)
+        {
+            Organisation organisation = dataRepository.Get<Organisation>(id);
+
+            var returns = organisation.Returns.OrderByDescending(r => r.AccountingDate).ThenByDescending(r => r.StatusDate);
+
+            var records = returns.Select(
+                ret =>
+                    new
+                    {
+                        OrganisationId = ret.Organisation.OrganisationId,
+                        OrganisationName = ret.Organisation.OrganisationName,
+                        ReturnId = ret.ReturnId,
+
+                        SnapshotDate = ret.AccountingDate,
+                        DeadlineDate = ret.AccountingDate.AddYears(1).AddDays(-1),
+                        ModifiedDate = ret.Modified,
+
+                        Status = ret.Status,
+                        Modifications = ret.Modifications,
+                        Late = ret.IsLateSubmission,
+                        LateReason = ret.LateReason,
+
+                        Employees = ret.OrganisationSize.GetAttribute<DisplayAttribute>().Name,
+
+                        HourlyPayGapMean = ret.DiffMeanHourlyPayPercent,
+                        HourlyPayGapMedian = ret.DiffMedianHourlyPercent,
+
+                        BonusPayPercentMale = ret.MaleMedianBonusPayPercent,
+                        BonusPayPercentFemale = ret.FemaleMedianBonusPayPercent,
+                        BonusPayGapMean = ret.DiffMeanBonusPercent,
+                        BonusPayGapMedian = ret.DiffMedianBonusPercent,
+
+                        UpperQuarterPercentMale = ret.MaleUpperQuartilePayBand,
+                        UpperQuarterPercentFemale = ret.FemaleUpperQuartilePayBand,
+                        UpperMiddleQuarterPercentMale = ret.MaleUpperPayBand,
+                        UpperMiddleQuarterPercentFemale = ret.FemaleUpperPayBand,
+                        LowerMiddleQuarterPercentMale = ret.MaleMiddlePayBand,
+                        LowerMiddleQuarterPercentFemale = ret.FemaleMiddlePayBand,
+                        LowerQuarterPercentMale = ret.MaleLowerPayBand,
+                        LowerQuarterPercentFemale = ret.FemaleLowerPayBand,
+
+                        LinkToCompanyWebsite = ret.CompanyLinkToGPGInfo,
+                        SeniorResponsibleOfficer = ret.ResponsiblePerson
+                    });
+
+            string fileDownloadName = $"ReturnsForOrganisation-{organisation.OrganisationId}--{VirtualDateTime.Now:yyyy-MM-dd HH:mm}.csv";

```

Comment:
[Not important] Would it be more user friendly to use the organisation name here?

Reply:
That's a really nice idea :-)